src/easylink/configuration.py

```diff
@@ -171,7 +171,7 @@ def _get_schema(self) -> Optional[PipelineSchema]:
         errors = defaultdict(dict)
         # Try each schema until one is validated
         for schema in PIPELINE_SCHEMAS:
-            logs = schema.validate_step(self.pipeline)
+            logs = schema.validate_step(self.pipeline, self.input_data)
```

Comment:
part of validating the pipeline schema is validating certain parts of the pipeline against the input data keys.

Reply:
Has this just been accidentally missing this whole time or is it somehow specific to multiples?

Reply:
Multiples are the only situation in which we currently use the input data config, so this is new. It also makes sense to add it to the abstractmethod, though, since a multiple could theoretically exist in any part of the step hierarchy.

Reply:
Yeah, this feels a bit weird, but it makes sense. Requesting an input dataset that doesn't exist is a "syntactical" error that can be flagged at parse time, so we don't want to make the user wait any longer than necessary.